main.go

```diff
@@ -57,3 +45,47 @@ func main() {
 
 	ctrl.Wait()
 }
+
+func newSyncConsumer(cfg *MateConfig) (consumers.Consumer, error) {
+	// New returns a Consumer implementation.
+	var consumer consumers.Consumer
+	var err error
+	switch cfg.Consumer {
+	case "google":
+		consumer, err = consumers.NewGoogleDNS(cfg.GoogleZone, cfg.GoogleProject, cfg.GoogleRecordGroupID)
+	case "aws":
+		consumer, err = consumers.NewAWSConsumer(cfg.AWSRecordGroupID)
+	case "stdout":
+		consumer, err = consumers.NewStdout()
```

Comment:
if you want we can also change these functions to:
* `consumers.NewGoogleCloudDNSConsumer`
* `consumers.NewAWSRoute53Consumer`
* `consumers.NewStdoutConsumer`

to have unified and unambiguous naming.